src/EditorFeatures/Core/Formatting/FormatCommandHandler.Paste.cs

```diff
@@ -88,6 +88,7 @@ private void ExecuteCommandWorker(PasteCommandArgs args, SnapshotPoint? caretPos
             var trackingSpan = caretPosition.Value.Snapshot.CreateTrackingSpan(caretPosition.Value.Position, 0, SpanTrackingMode.EdgeInclusive);
             var span = trackingSpan.GetSpan(args.SubjectBuffer.CurrentSnapshot).Span.ToTextSpan();
 
+            // Note: C# always completes synchronously, Type Script is async
             var changes = formattingService.GetFormattingChangesOnPasteAsync(document, args.SubjectBuffer, span, cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
i wonder if we should add things like `SupportsSynchronous` to the api and `GetFormattingChangesOnPaste` to enforce some of this?

Reply:
I'd rather push the Wait to TypeScript and pretend on our end that everything is synchronous.